pandas/io/formats/printing.py

```diff
@@ -178,21 +177,13 @@ def pprint_thing(
 
     Returns
     -------
-    result - unicode str
+    str
 
     """
 
-    def as_escaped_unicode(thing, escape_chars=escape_chars):
-        # Unicode is fine, else we try to decode using utf-8 and 'replace'
-        # if that's not it either, we have no way of knowing and the user
-        # should deal with it himself.
-
-        try:
-            result = str(thing)  # we should try this first
-        except UnicodeDecodeError:
-            # either utf-8 or we replace errors
-            result = str(thing).decode("utf-8", "replace")
-
+    def as_escaped_string(
+        thing: object, escape_chars: Optional[EscapeChars] = escape_chars
```

Comment:
Can you use `Any` instead of `object`? Won't change output since this is all green but would be more idiomatic

Reply:
Contrast the behavior of Any with the behavior of object. Similar to Any, every type is a subtype of object. However, unlike Any, the reverse is not true: object is not a subtype of every other type.

That means when the type of a value is object, a type checker will reject almost all operations on it, and assigning it to a variable (or using it as a return value) of a more specialized type is a type error.

Use object to indicate that a value could be any type in a typesafe manner. Use Any to indicate that a value is dynamically typed.

https://docs.python.org/3/library/typing.html#the-any-type

object is used here to indicate that a value could be any type in a typesafe manner and reject almost all operations on it.

Reply:
In the outer function, thing should be Any since it is dynamically typed. we used hasattr introspection. But following previous discussions, refraining from using Any and just leaving untyped.